base/sort.jl

```diff
@@ -288,14 +293,24 @@ for s in [:searchsortedfirst, :searchsortedlast, :searchsorted]
 end
 
 """
-    searchsorted(a, x; by=<transform>, lt=<comparison>, rev=false)
+    searchsorted(v, x; by=identity, lt=isless, rev=false)
+
+Return the range of indices in `v` where values are equivalent to `x`, or an
+empty range located at the insertion point if `v` does not contain values
+equivalent to `x` (see below for the definition of equivalence).
+
+The range is found using binary search. The vector `v` must be sorted, or at
+least partitioned with respect to `x` such that all values that sort before `x`
+come first and all values that sort after `x` come last.
+
+The `by`, `lt` and `rev` keywords modify what order is assumed for the data,
+as described in the [`sort!`](@ref) documentation.
 
-Return the range of indices of `a` which compare as equal to `x` (using binary search)
-according to the order specified by the `by`, `lt` and `rev` keywords, assuming that `a`
-is already sorted in that order. Return an empty range located at the insertion point
-if `a` does not contain values equal to `x`.
+Values `x` and `y` are said equivalent if `lt(by(x), by(y))` and `lt(by(y),
+by(x))` both return `false`, and `x` is said to sort before `y` if `lt(by(x),
+by(y))` returns `true` (with `x` and `y` exchanged if `rev=true`).
```

Comment:
It would be nice to refer to an existing canonical definition of orderings and define equivalence to be "niegher is less than the other" rather than reiterate the whole thing in a different wording here.

Reply:
Good idea. I made some changes, now the various concepts (less, greater and equivalence) are precisely defined in `sort!` and other docstrings refer to that.